### PR TITLE
Fix missing step in guidance

### LINKED ISSUE #3412: Gather Asset Manifests still runs with YAML stages unless global _PublishUsingPipelines variable is true
The docs for stages mentions that you just need to set `enablePublishUsingPipelines: true` if you use the jobs.yml template and only use `_PublishUsingPipelines` if you don't: https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/YamlStagesPublishing.md#how-to-onboard-onto-yaml-based-publishing

However the Gather Asset Manifests task only checks for the `_PublishUsingPipelines` variable so if you don't have the variable set it will still run: https://github.com/dotnet/arcade/blob/316481e57ee5e6acbbf2401eb6778a1d3d48d25b/eng/common/templates/job/job.yml#L190

/cc @riarenas @jcagme

### PR DESCRIPTION
**Pull request by riarenas**
Fixes #3412

### CODE REVIEW

File: Documentation/CorePackages/YamlStagesPublishing.md [hunk 1]
```diff
@@ -24,18 +24,7 @@ In order to use this new publishing mechanism, the easiest way to start is by ma
...
-    jobs:
-    - template: /eng/common/templates/jobs/jobs.yml
-      parameters:
-        enablePublishUsingPipelines: true
-    ```
-
-    It is recommended to use the jobs.yml template to manage this property, as it will make sure to flow it to all the jobs and steps that require it.  The template also handles a lot of the boilerplate steps that need to be performed during the build, such as the publishing of build assets to the BAR. More information about the jobs.yml template can be found [here](../AzureDevOps/PhaseToJobSchemaChange.md#--what-is-the-engcommontemplatesjobsjobsyml-template).  If it is not possible for the repo to use the jobs.yml template, the following steps need to be performed:
-
-    * Set a variable called `_PublishUsingPipelines` with a value of `true` so that the gathering of asset manifests won't be performed during the build.
+    Set a variable called `_PublishUsingPipelines` with a value of `true` so that the gathering of asset manifests won't be performed during the build.
```
Comment (akoeplinger) [inline Documentation/CorePackages/YamlStagesPublishing.md]: In my experiments it didn't work for some reason if I set it as a global variable, I had to add it to the variables section of the jobs.yml template instead:
```
  - template: /eng/common/templates/jobs/jobs.yml
    parameters:
      enablePublishUsingPipelines: true
      enablePublishBuildAssets: true
      variables:
        - _PublishUsingPipelines: true
```

Comment (JohnTortugo) [reply Documentation/CorePackages/YamlStagesPublishing.md]: @akoeplinger can you point me to the build-def/branch where you tested it?

Comment (akoeplinger) [reply Documentation/CorePackages/YamlStagesPublishing.md]: @JohnTortugo https://dev.azure.com/dnceng/internal/_build/results?buildId=281742&view=logs&s=6884a131-87da-5381-61f3-d7acc3b91d76&j=77530c68-e4ad-52fb-cdbc-d06cd1e95a78 was the build for this yaml: https://github.com/mono/mono/commit/882208bbc602917b1c951a349cd07fa57a9f029b/

As you can see it still runs the `Gather Asset Manifests` step.

**Comment by riarenas** [comment]
@johntortugo is working instead of passing the parameter through to job.yml instead of relying on a global variable. Let's go with that approach instead.

File: Documentation/CorePackages/YamlStagesPublishing.md (hunk 1 above)
Comment (JohnTortugo) [reply Documentation/CorePackages/YamlStagesPublishing.md]: @akoeplinger thanks for the links. I've a PR in progress to fix the YAML template: https://github.com/dotnet/arcade/pull/3482 . I'll also work on updating the documentation and ping you after I get it done. I'll do it tomorrow.

**Pull request closed without merging**